src/apps/backend/modules/account/account-service.ts

```diff
@@ -22,4 +31,52 @@ export default class AccountService {
   ): Promise<Account> {
     return AccountReader.getAccountById(params.accountId);
   }
+
+  public static async getAccountByContactNumber(
+    params: AccountSearchParams,
+  ): Promise<Account> {
+    return AccountReader.getAccountByContactNumber(params.contactNumber);
+  }
+
+  public static async createAccountWithContactNumber(
+    params: CreateAccountParams,
+  ): Promise<Account> {
+    return AccountWriter.createAccountWithContactNumber(params);
+  }
+
+  public static async sendOTP(contactNumber: string): Promise<Account> {
+    await SMSService.sendOTP(contactNumber);
+
+    const params: CreateAccountParams = {
+      contactNumber,
+      firstName: null,
+      lastName: null,
+      password: null,
+      username: null,
+    };
+
+    const account = await AccountReader.getAccountByContactNumber(params.contactNumber);
+
+    if (account) {
```

Comment:
There is one use case where I need to check if the user exists, and if it doesn't, then I create that user while sending an OTP.
Previously, it was throwing an error when the user wasn't found by the contact number, which would stop us from creating the user.

Reply:
I understand, but your code is not right.

The function `AccountReader.getAccountByContactNumber(params.contactNumber);` should be `AccountReader.getAccountByContactNumberOptional(params.contactNumber);` It should also return `Promise<Nullable<Account>>`  instead of `Promise<Account>`

`Nullable<T> = T | null | undefined` probably